src/util.c

```diff
@@ -198,6 +198,43 @@ static int stringmatchlen_impl(const char *pattern, int patternLen,
     return 0;
 }
 
+/* 
+ * glob-style pattern matching to check if a given pattern fully includes 
+ * the prefix of a string. For the match to succeed, the pattern must end with 
+ * an unescaped '*' character.
+ * 
+ * Returns: 1 if the `pattern` fully matches the `prefixStr`. Returns 0 otherwise.
+ */
+int prefixmatch(const char *pattern, int patternLen,
```

Comment:
adding some unittests for this method?

Reply:
I think it is sufficiently covered by TCL. pls lmk otherwise.

Reply:
Do we just always ignore the https://en.wikibooks.org/wiki/C_Programming/stdbool.h? Why? This function only returns two values out of the whole range of ints, yet it asks the compiler to reserve the whole int range and the corresponding register, also removing the chances of possible return value optimisations and inlining the calls to it (of course, the compiler might still decide to do that after the complete analysis stage, but this requires to compile time, and a good compiler, which we don't always have (we still build sometimes using very old compiler versions)), not to mention that for the reader seeing an `int` being returned means, there might be a range of error codes, but there is none, nothing else rather than `0` and `1`. Why not just use the provided boolean type from C99? What is the negative impact of using `bool` and what is the absolutely positive side of using an `int` here?

Reply:
Also, just following the C principles, as well as UNIX, `0` means **good** or **correct**, anything else - otherwise. Why is the logic inverted here?

Reply:
It is considered as bool and aligned with util.c conventions. Specifically aligned with `stringmatch`.

Reply:
Ooh, okay, thank you, got it.